web-server/pages/api/resources/orgs/[org_id]/teams/v2.ts

```diff
@@ -302,3 +308,37 @@ const updateReposWorkflows = async (
     }
   }
 };
+
+const getTeamReposMap = async (org_id: ID, providers: Integration[]) => {
+  const dbRepos: RepoWithMultipleWorkflows[] = await db(Table.OrgRepo)
+    .leftJoin({ tr: Table.TeamRepos }, function () {
+      this.on('OrgRepo.id', 'tr.org_repo_id');
+    })
+    .where('tr.is_active', true)
+    .leftJoin({ rw: Table.RepoWorkflow }, function () {
+      this.on('OrgRepo.id', 'rw.org_repo_id').andOn(
+        'rw.is_active',
+        '=',
+        dbRaw.raw(true)
+      );
+    })
+    .select('OrgRepo.*')
+    .select(
+      dbRaw.raw(
+        "COALESCE(json_agg(json_build_object('name', rw.name, 'value', rw.provider_workflow_id)) FILTER (WHERE rw.id IS NOT NULL), '[]') as repo_workflows"
+      )
+    )
+    .select('tr.deployment_type', 'tr.team_id')
+    .from('OrgRepo')
+    .where('org_id', org_id)
+    .and.whereIn('OrgRepo.provider', providers)
+    .andWhereNot('OrgRepo.is_active', false)
+    .groupBy(
+      'OrgRepo.id',
+      'rw.org_repo_id',
+      'tr.deployment_type',
+      'tr.team_id'
+    );
```

Comment:
This is pretty hard to read.
1. Let's try to simplify this for readability. Let me know if you'd like suggestions or give it a shot on your own instead.
2. Can we replace the .raw usage in some form? Or at least reduce the amount of stuff supplied to it?

Reply:
Query is now more readable and provide enough data.

```javascript
const dbRepos: RepoWithMultipleWorkflows[] = await db(Table.OrgRepo)
  .where('org_id', org_id)
  .and.whereIn('OrgRepo.provider', providers)
  .andWhereNot('OrgRepo.is_active', false)
  .leftJoin({ tr: Table.TeamRepos }, 'OrgRepo.id', 'tr.org_repo_id')
  .andWhere('tr.is_active', true)
  .leftJoin({ rw: Table.RepoWorkflow }, 'OrgRepo.id', 'rw.org_repo_id')
  .andWhere('rw.is_active', true)
  .select('OrgRepo.*')
  .select(dbRaw.raw('json_agg(rw.*) as repo_workflows'))
  .select('tr.deployment_type', 'tr.team_id')
  .groupBy('OrgRepo.id', 'tr.deployment_type', 'tr.team_id');

const repoWithWorkflows = dbRepos.map((repo) => {
  const updatedWorkflows = repo.repo_workflows.map(
    (workflow: RepoWorkflow) => ({
      name: workflow.name,
      value: workflow.provider_workflow_id
    })
  );
  return {
    ...repo,
    repo_workflows: updatedWorkflows
  };
});
```

Reply:
I think is_active in OrgRepo doesn't have any significance.

Reply:
Does this looks better?

```javascript
const dbRepos: RepoWithMultipleWorkflows[] = await db(Table.OrgRepo)
  .where('org_id', org_id)
  .and.whereIn('OrgRepo.provider', providers)
  .where('OrgRepo.is_active', true)
  .leftJoin({ tr: Table.TeamRepos }, 'OrgRepo.id', 'tr.org_repo_id')
  .andWhere('tr.is_active', true)
  .leftJoin({ rw: Table.RepoWorkflow }, 'OrgRepo.id', 'rw.org_repo_id')
  .andWhere(function () {
    this.where('rw.is_active', true).orWhereNull('rw.is_active');
  })
  .select('OrgRepo.*')
  .select(
    dbRaw.raw(
      "COALESCE(json_agg(rw.*) FILTER (WHERE rw.id IS NOT NULL), '[]') as repo_workflows"
    )
  )
  .select('tr.deployment_type', 'tr.team_id')
  .groupBy('OrgRepo.id', 'tr.deployment_type', 'tr.team_id');

const repoWithWorkflows = dbRepos.map((repo) => {
  const updatedWorkflows = repo.repo_workflows.map(
    (workflow: RepoWorkflow) => ({
      name: workflow.name,
      value: workflow.provider_workflow_id
    })
  );
  return {
    ...repo,
    repo_workflows: updatedWorkflows
  };
});
```

Reply:
I think is_active have no significance in OrgRepo table.

Reply:
`is_active` in the OrgRepo table determines whether the repo still needs to be synced with Middleware or not.
It's pretty important. 😄

Reply:
I made a refactor that I think helps me wrap my head around the code a lot better:

```ts
const baseQuery = db(Table.OrgRepo)
  .where('org_id', org_id)
  .andWhere('OrgRepo.is_active', true)
  .whereIn('OrgRepo.provider', providers);

const teamJoin = baseQuery
  .leftJoin({ tr: Table.TeamRepos }, 'OrgRepo.id', 'tr.org_repo_id')
  .andWhere('tr.is_active', true);

const workflowJoin = teamJoin
  .leftJoin({ rw: Table.RepoWorkflow }, 'OrgRepo.id', 'rw.org_repo_id')
  .andWhere('rw.is_active', true)
  .orWhereNull('rw.is_active');

const dbRepos = await workflowJoin
  .select('OrgRepo.*')
  .select(
    db('RepoWorkflow')
      .select(db.raw("COALESCE(json_agg(rw.*), '[]')"))
      .whereNotNull('rw.id')
      .as('repo_workflows')
  )
  .select('tr.deployment_type', 'tr.team_id')
  .groupBy('OrgRepo.id', 'tr.deployment_type', 'tr.team_id');

const repoWithWorkflows = dbRepos.map((repo) => {
  const updatedWorkflows = repo.repo_workflows.map((workflow) => ({
    name: workflow.name,
    value: workflow.provider_workflow_id,
  }));

  return {
    ...repo,
    repo_workflows: updatedWorkflows,
  };
});
```

Reply:
Awesome ✨ ✨

Reply:
Oh, I see that if all TeamRepos are inactive, the corresponding OrgRepo will be marked inactive.

However if a team is deleted, nor the corresponding TeamRepo or OrgRepo is marked inactive.

Reply:
Good catch. That should get handled.